src/registry.rs

```diff
@@ -73,7 +73,7 @@ pub fn get_registry_with_config(config: Configuration) -> &'static Registry {
 /// function. Declared `unsafe` because it writes to `THE_REGISTRY` in
 /// an unsynchronized fashion.
 unsafe fn init_registry(config: Configuration) {
-    let registry = leak(Arc::new(Registry::new(config)));
+    let registry = leak(Registry::new(config));
     THE_REGISTRY = Some(registry);
```

Comment:
Huh.  Note that `Registry::new` returns `Arc<Registry>`, so after `leak` it was `registry: &'static Arc<Arc<Registry>>`.  I guess some auto-deref magic was letting this be written to `THE_REGISTRY: Option<&'static Arc<Registry>>`.

Now with your change, we have `registry: &'static Arc<Registry>` writing to `THE_REGISTRY: Option<&'static Registry>`, so there's still some auto-deref going on in forming `Some(registry)`.

I wonder, why shouldn't we just forget `fn leak` and use `THE_REGISTRY: Option<Arc<Registry>>` ?